scripts/train.py

```diff
@@ -0,0 +1,178 @@
+config_full_path = os.path.join(root_full_path, 'config/config.json')
+if not os.path.exists(config_full_path):
+    print(f"{config_full_path} doesn't exist.")
+    exit(0)
+with open(config_full_path, 'r') as file:
+    top_config = json.load(file)
+os.chdir(cwd)
+
+# import local functions
+from nn_modules.transformer.top_former import TopFormer
+from data.data_loader import train_data_loader, validation_data_loader
+from utils.helper import count_parameters, initialize_weights, epoch_time
+
+model = TopFormer(
+    num_layers=top_config["model_params"]["num_layers"],
+    d_model=top_config["model_params"]["d_model"],
+    nhead=top_config["model_params"]["nhead"],
+    num_classes=top_config["model_params"]["num_classes"])
+
+
+# TODO(diwei): Remove this hint
+print(f'The model has {count_parameters(model)}: trainable parameters')
+model.apply(initialize_weights)
+
+
+# TODO(diwei): Adjust the type and parameters of the optimizer
+optimizer = Adam(params=model.parameters(),
+                 lr=top_config["train_params"]["optimize_params"]["lr"],
+                 weight_decay=top_config["train_params"]["optimize_params"]["weight_decay"],
+                 eps=top_config["train_params"]["optimize_params"]["eps"])
+
+scheduler = optim.lr_scheduler.ReduceLROnPlateau(optimizer=optimizer,
+                                                 verbose=True)
+
+# TODO(diwei): Adjust the type and parameters of the loss
+loss = nn.CrossEntropyLoss()
+
+
+def train(model, data_loader, optimizer, loss, device='cpu'):
+    """The interface to train the model using training data.
+
+    Parameters
+    ----------
+    model : nn.Module
+        The model to be trained.
+    data_loader : DataLoader
+        The interface for accessing training data and its corresponding labels 
+        as the ground truth.
+    optimizer : Adam
+        The optimization algorithm for updating model parameters.
+    loss : Loss nn.CrossEntropyLoss
+        The loss function as objective and constraint for learning model parameters.
+    device : str, optional
+        The computing device for training the model (default: 'cpu').
+
+    Returns
+    -------
+    float
+        The loss average of the entire data and update the learninf model parameters.
+    """
+    model.train()
+    epoch_loss = 0
+    model.to(device)
+    for i, batch in enumerate(data_loader):
+        tgt = batch["tgt"]
+        mem = batch["mem"]
+        gt = batch["gt"]
+
+        optimizer.zero_grad()
+        output = model(tgt.permute(1, 0, 2), mem.permute(1, 0, 2))
+        loss_result = loss(torch.max(output, dim=0)[0], gt)
+        loss_result.backward()
+        optimizer.step()
+
+        epoch_loss += loss_result.item()
+
+        # TODO(diwei): use other progressbar
+        print('step :', round((i / (len(data_loader) + + sys.float_info.epsilon)) * 100, 2),
+              '% , loss :', loss_result.item())
+
+    return epoch_loss / (len(data_loader) + sys.float_info.epsilon)
```

Comment:
The denominator should be an integer value, so I suggest we do not use an epsilon here but throw an error if the denominator becomes zero.